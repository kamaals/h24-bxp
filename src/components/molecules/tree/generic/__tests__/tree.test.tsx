import { describe, it, vi, expect, beforeEach } from "vitest";
import { render, screen, cleanup } from "@testing-library/react";
import { Tree } from "../index";
import {
  LeafComponent,
  TitleLeafComponent,
} from "@/components/molecules/tree/generic/leaf";

const onClick = vi.fn();
const expandClick = vi.fn();

const data = [
  {
    id: "1",
    name: "Test",
    onClick: expandClick,
    children: [
      {
        id: "1.1",
        name: "Test 1.1",
      },
    ],
  },
  {
    id: "2",
    name: "Test 2",
    children: [
      {
        id: "2.1",
        name: "Test 2.1",
      },
    ],
  },
  {
    id: "3",
    name: "Test 3",
    onClick,
  },
];

describe(Tree.name, () => {
  beforeEach(() => {
    cleanup();
  });

  it("Should trigger the onClick function", () => {
    render(<Tree data={data} />);
    const item = screen.getByTestId("tree-leaf-3");
    item.click();
    expect(onClick).toHaveBeenCalledWith(data[2]);
  });

  it("Should trigger the onClick function on expandable", () => {
    render(<Tree data={data} />);
    const item = screen.getByTestId("tree-trigger-1");
    item.click();
    expect(onClick).toHaveBeenCalledWith(data[2]);
  });

  it("Should match the snapshot", () => {
    const resp = render(
      <Tree
        data={data}
        initialSelectedItemId={"1"}
        titleLeaf={TitleLeafComponent}
        leaf={LeafComponent}
      />,
    );
    expect(resp).toMatchSnapshot();
  });
});
