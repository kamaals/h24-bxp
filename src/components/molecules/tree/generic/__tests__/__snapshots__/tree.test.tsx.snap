// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Tree > Should match the snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class=""
      >
        <ul>
          <li
            class="horizontal-line after:absolute after:rounded-bl-lg after:border-l after:left-1 after:top-3.5 after:w-4 after:h-4 after:border-b after:z-100 relative vertical-line before:absolute before:left-1 before:top-0 before:bottom-0 before:w-2 before:border-l before:z-100 [&:last-child]:before:h-4 pt-2"
          >
            <div
              data-orientation="vertical"
            >
              <div
                class="pl-4 ml-1 group"
                data-orientation="vertical"
                data-state="open"
              >
                <div
                  class="flex relative"
                >
                  <div
                    class="flex-grow flex items-start h-11 p-0.5 rounded-lg relative transition-left duration-200 after:absolute after:top-11 after:bottom-0 after:left-4 after:w-1 after:border-l group-data-[state=closed]:after:border-l-0"
                  >
                    <div
                      class="pointer-events-none absolute inset-0 size-full rounded-[inherit] will-change-[background-position] motion-safe:animate-shine"
                      style="--border-width: 2px; --duration: 14s; background-size: 300% 300%; mask: linear-gradient(#fff 0 0) content-box, linear-gradient(#fff 0 0); mask-composite: exclude;"
                    />
                    <h3
                      class="bg-zinc-50 h-10 flex items-center border rounded-l-lg border-r-0"
                      data-orientation="vertical"
                      data-state="open"
                    >
                      <button
                        aria-controls="radix-«r9»"
                        aria-expanded="true"
                        class="accordion-trigger flex items-center transition-all first:[&[data-state=open]>svg]:rotate-90 before:opacity-100 before:bg-white text-accent-foreground before:z-10 has-child before:left-1 flex-grow h-10 2 cursor-pointer"
                        data-orientation="vertical"
                        data-radix-collection-item=""
                        data-state="open"
                        data-testid="tree-trigger-1"
                        id="radix-«r8»"
                        type="button"
                      >
                        <svg
                          aria-hidden="true"
                          class="lucide lucide-chevron-right h-4 w-4 shrink-0 ml-2 transition-transform duration-200 text-accent-foreground/50 mr-1 relative z-20"
                          fill="none"
                          height="24"
                          stroke="currentColor"
                          stroke-linecap="round"
                          stroke-linejoin="round"
                          stroke-width="2"
                          viewBox="0 0 24 24"
                          width="24"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="m9 18 6-6-6-6"
                          />
                        </svg>
                      </button>
                    </h3>
                    <div
                      class="flex-1 border-r rounded-r-lg overflow-hidden"
                    >
                      <div
                        class="text-sm truncate relative z-20 bg-zinc-50 h-10 border-b border-t flex items-center"
                      >
                        <span>
                          Test
                        </span>
                      </div>
                    </div>
                  </div>
                  <div
                    class="flex absolute items-center h-10 right-0 z-20"
                    data-testid="tree-actions-undefined"
                  />
                </div>
                <div
                  aria-labelledby="radix-«r8»"
                  class="overflow-hidden text-sm transition-all data-[state=closed]:animate-accordion-up data-[state=open]:animate-accordion-down ml-3"
                  data-orientation="vertical"
                  data-state="open"
                  id="radix-«r9»"
                  role="region"
                  style="--radix-accordion-content-height: var(--radix-collapsible-content-height); --radix-accordion-content-width: var(--radix-collapsible-content-width); transition-duration: 0s; animation-name: none;"
                >
                  <div
                    class="pt-0"
                  >
                    <div
                      class=""
                    >
                      <ul>
                        <li
                          class="horizontal-line after:absolute after:rounded-bl-lg after:border-l after:left-1 after:top-3.5 after:h-4 after:border-b after:z-100 after:w-4 relative vertical-line before:absolute before:left-1 before:top-0 before:bottom-0 before:w-2 before:border-l before:z-100 [&:last-child]:before:h-4 pt-2"
                        >
                          <div
                            class="ml-5 h-10 border rounded-lg bg-zinc-50 flex text-left items-center p-2"
                          >
                            <div
                              class="flex flex-1 items-center cursor-pointer relative group hover:before:opacity-100 before:absolute before:rounded-lg before:-left-0 before:right-0 before:bg-accent before:opacity-0 before:h-[2rem] before:-z-10"
                              data-testid="tree-leaf-1.1"
                            >
                              <span
                                class="flex-grow text-sm truncate"
                              >
                                Test 1.1
                              </span>
                            </div>
                            <div
                              class="hidden absolute items-center h-10 right-0 z-20"
                              data-testid="tree-actions-1.1"
                            />
                          </div>
                        </li>
                      </ul>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </li>
          <li
            class="horizontal-line after:absolute after:rounded-bl-lg after:border-l after:left-1 after:top-3.5 after:w-4 after:h-4 after:border-b after:z-100 relative vertical-line before:absolute before:left-1 before:top-0 before:bottom-0 before:w-2 before:border-l before:z-100 [&:last-child]:before:h-4 pt-2"
          >
            <div
              data-orientation="vertical"
            >
              <div
                class="pl-4 ml-1 group"
                data-orientation="vertical"
                data-state="closed"
              >
                <div
                  class="flex relative"
                >
                  <div
                    class="flex-grow flex items-start h-11 p-0.5 rounded-lg relative transition-left duration-200 after:absolute after:top-11 after:bottom-0 after:left-4 after:w-1 after:border-l group-data-[state=closed]:after:border-l-0"
                  >
                    <h3
                      class="bg-zinc-50 h-10 flex items-center border rounded-l-lg border-r-0"
                      data-orientation="vertical"
                      data-state="closed"
                    >
                      <button
                        aria-controls="radix-«rb»"
                        aria-expanded="false"
                        class="accordion-trigger flex items-center transition-all first:[&[data-state=open]>svg]:rotate-90 has-child before:left-1 flex-grow h-10 2 cursor-pointer"
                        data-orientation="vertical"
                        data-radix-collection-item=""
                        data-state="closed"
                        data-testid="tree-trigger-2"
                        id="radix-«ra»"
                        type="button"
                      >
                        <svg
                          aria-hidden="true"
                          class="lucide lucide-chevron-right h-4 w-4 shrink-0 ml-2 transition-transform duration-200 text-accent-foreground/50 mr-1 relative z-20"
                          fill="none"
                          height="24"
                          stroke="currentColor"
                          stroke-linecap="round"
                          stroke-linejoin="round"
                          stroke-width="2"
                          viewBox="0 0 24 24"
                          width="24"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="m9 18 6-6-6-6"
                          />
                        </svg>
                      </button>
                    </h3>
                    <div
                      class="flex-1 border-r rounded-r-lg overflow-hidden"
                    >
                      <div
                        class="text-sm truncate relative z-20 bg-zinc-50 h-10 border-b border-t flex items-center"
                      >
                        <span>
                          Test 2
                        </span>
                      </div>
                    </div>
                  </div>
                  <div
                    class="hidden absolute items-center h-10 right-0 z-20"
                    data-testid="tree-actions-undefined"
                  />
                </div>
                <div
                  aria-labelledby="radix-«ra»"
                  class="overflow-hidden text-sm transition-all data-[state=closed]:animate-accordion-up data-[state=open]:animate-accordion-down ml-3"
                  data-orientation="vertical"
                  data-state="closed"
                  hidden=""
                  id="radix-«rb»"
                  role="region"
                  style="--radix-accordion-content-height: var(--radix-collapsible-content-height); --radix-accordion-content-width: var(--radix-collapsible-content-width);"
                />
              </div>
            </div>
          </li>
          <li
            class="horizontal-line after:absolute after:rounded-bl-lg after:border-l after:left-1 after:top-3.5 after:h-4 after:border-b after:z-100 after:w-4 relative vertical-line before:absolute before:left-1 before:top-0 before:bottom-0 before:w-2 before:border-l before:z-100 [&:last-child]:before:h-4 pt-2"
          >
            <div
              class="ml-5 h-10 border rounded-lg bg-zinc-50 flex text-left items-center p-2"
            >
              <div
                class="flex flex-1 items-center cursor-pointer relative group hover:before:opacity-100 before:absolute before:rounded-lg before:-left-0 before:right-0 before:bg-accent before:opacity-0 before:h-[2rem] before:-z-10"
                data-testid="tree-leaf-3"
              >
                <span
                  class="flex-grow text-sm truncate"
                >
                  Test 3
                </span>
              </div>
              <div
                class="hidden absolute items-center h-10 right-0 z-20"
                data-testid="tree-actions-3"
              />
            </div>
          </li>
        </ul>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class=""
    >
      <ul>
        <li
          class="horizontal-line after:absolute after:rounded-bl-lg after:border-l after:left-1 after:top-3.5 after:w-4 after:h-4 after:border-b after:z-100 relative vertical-line before:absolute before:left-1 before:top-0 before:bottom-0 before:w-2 before:border-l before:z-100 [&:last-child]:before:h-4 pt-2"
        >
          <div
            data-orientation="vertical"
          >
            <div
              class="pl-4 ml-1 group"
              data-orientation="vertical"
              data-state="open"
            >
              <div
                class="flex relative"
              >
                <div
                  class="flex-grow flex items-start h-11 p-0.5 rounded-lg relative transition-left duration-200 after:absolute after:top-11 after:bottom-0 after:left-4 after:w-1 after:border-l group-data-[state=closed]:after:border-l-0"
                >
                  <div
                    class="pointer-events-none absolute inset-0 size-full rounded-[inherit] will-change-[background-position] motion-safe:animate-shine"
                    style="--border-width: 2px; --duration: 14s; background-size: 300% 300%; mask: linear-gradient(#fff 0 0) content-box, linear-gradient(#fff 0 0); mask-composite: exclude;"
                  />
                  <h3
                    class="bg-zinc-50 h-10 flex items-center border rounded-l-lg border-r-0"
                    data-orientation="vertical"
                    data-state="open"
                  >
                    <button
                      aria-controls="radix-«r9»"
                      aria-expanded="true"
                      class="accordion-trigger flex items-center transition-all first:[&[data-state=open]>svg]:rotate-90 before:opacity-100 before:bg-white text-accent-foreground before:z-10 has-child before:left-1 flex-grow h-10 2 cursor-pointer"
                      data-orientation="vertical"
                      data-radix-collection-item=""
                      data-state="open"
                      data-testid="tree-trigger-1"
                      id="radix-«r8»"
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        class="lucide lucide-chevron-right h-4 w-4 shrink-0 ml-2 transition-transform duration-200 text-accent-foreground/50 mr-1 relative z-20"
                        fill="none"
                        height="24"
                        stroke="currentColor"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                        stroke-width="2"
                        viewBox="0 0 24 24"
                        width="24"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="m9 18 6-6-6-6"
                        />
                      </svg>
                    </button>
                  </h3>
                  <div
                    class="flex-1 border-r rounded-r-lg overflow-hidden"
                  >
                    <div
                      class="text-sm truncate relative z-20 bg-zinc-50 h-10 border-b border-t flex items-center"
                    >
                      <span>
                        Test
                      </span>
                    </div>
                  </div>
                </div>
                <div
                  class="flex absolute items-center h-10 right-0 z-20"
                  data-testid="tree-actions-undefined"
                />
              </div>
              <div
                aria-labelledby="radix-«r8»"
                class="overflow-hidden text-sm transition-all data-[state=closed]:animate-accordion-up data-[state=open]:animate-accordion-down ml-3"
                data-orientation="vertical"
                data-state="open"
                id="radix-«r9»"
                role="region"
                style="--radix-accordion-content-height: var(--radix-collapsible-content-height); --radix-accordion-content-width: var(--radix-collapsible-content-width); transition-duration: 0s; animation-name: none;"
              >
                <div
                  class="pt-0"
                >
                  <div
                    class=""
                  >
                    <ul>
                      <li
                        class="horizontal-line after:absolute after:rounded-bl-lg after:border-l after:left-1 after:top-3.5 after:h-4 after:border-b after:z-100 after:w-4 relative vertical-line before:absolute before:left-1 before:top-0 before:bottom-0 before:w-2 before:border-l before:z-100 [&:last-child]:before:h-4 pt-2"
                      >
                        <div
                          class="ml-5 h-10 border rounded-lg bg-zinc-50 flex text-left items-center p-2"
                        >
                          <div
                            class="flex flex-1 items-center cursor-pointer relative group hover:before:opacity-100 before:absolute before:rounded-lg before:-left-0 before:right-0 before:bg-accent before:opacity-0 before:h-[2rem] before:-z-10"
                            data-testid="tree-leaf-1.1"
                          >
                            <span
                              class="flex-grow text-sm truncate"
                            >
                              Test 1.1
                            </span>
                          </div>
                          <div
                            class="hidden absolute items-center h-10 right-0 z-20"
                            data-testid="tree-actions-1.1"
                          />
                        </div>
                      </li>
                    </ul>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </li>
        <li
          class="horizontal-line after:absolute after:rounded-bl-lg after:border-l after:left-1 after:top-3.5 after:w-4 after:h-4 after:border-b after:z-100 relative vertical-line before:absolute before:left-1 before:top-0 before:bottom-0 before:w-2 before:border-l before:z-100 [&:last-child]:before:h-4 pt-2"
        >
          <div
            data-orientation="vertical"
          >
            <div
              class="pl-4 ml-1 group"
              data-orientation="vertical"
              data-state="closed"
            >
              <div
                class="flex relative"
              >
                <div
                  class="flex-grow flex items-start h-11 p-0.5 rounded-lg relative transition-left duration-200 after:absolute after:top-11 after:bottom-0 after:left-4 after:w-1 after:border-l group-data-[state=closed]:after:border-l-0"
                >
                  <h3
                    class="bg-zinc-50 h-10 flex items-center border rounded-l-lg border-r-0"
                    data-orientation="vertical"
                    data-state="closed"
                  >
                    <button
                      aria-controls="radix-«rb»"
                      aria-expanded="false"
                      class="accordion-trigger flex items-center transition-all first:[&[data-state=open]>svg]:rotate-90 has-child before:left-1 flex-grow h-10 2 cursor-pointer"
                      data-orientation="vertical"
                      data-radix-collection-item=""
                      data-state="closed"
                      data-testid="tree-trigger-2"
                      id="radix-«ra»"
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        class="lucide lucide-chevron-right h-4 w-4 shrink-0 ml-2 transition-transform duration-200 text-accent-foreground/50 mr-1 relative z-20"
                        fill="none"
                        height="24"
                        stroke="currentColor"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                        stroke-width="2"
                        viewBox="0 0 24 24"
                        width="24"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="m9 18 6-6-6-6"
                        />
                      </svg>
                    </button>
                  </h3>
                  <div
                    class="flex-1 border-r rounded-r-lg overflow-hidden"
                  >
                    <div
                      class="text-sm truncate relative z-20 bg-zinc-50 h-10 border-b border-t flex items-center"
                    >
                      <span>
                        Test 2
                      </span>
                    </div>
                  </div>
                </div>
                <div
                  class="hidden absolute items-center h-10 right-0 z-20"
                  data-testid="tree-actions-undefined"
                />
              </div>
              <div
                aria-labelledby="radix-«ra»"
                class="overflow-hidden text-sm transition-all data-[state=closed]:animate-accordion-up data-[state=open]:animate-accordion-down ml-3"
                data-orientation="vertical"
                data-state="closed"
                hidden=""
                id="radix-«rb»"
                role="region"
                style="--radix-accordion-content-height: var(--radix-collapsible-content-height); --radix-accordion-content-width: var(--radix-collapsible-content-width);"
              />
            </div>
          </div>
        </li>
        <li
          class="horizontal-line after:absolute after:rounded-bl-lg after:border-l after:left-1 after:top-3.5 after:h-4 after:border-b after:z-100 after:w-4 relative vertical-line before:absolute before:left-1 before:top-0 before:bottom-0 before:w-2 before:border-l before:z-100 [&:last-child]:before:h-4 pt-2"
        >
          <div
            class="ml-5 h-10 border rounded-lg bg-zinc-50 flex text-left items-center p-2"
          >
            <div
              class="flex flex-1 items-center cursor-pointer relative group hover:before:opacity-100 before:absolute before:rounded-lg before:-left-0 before:right-0 before:bg-accent before:opacity-0 before:h-[2rem] before:-z-10"
              data-testid="tree-leaf-3"
            >
              <span
                class="flex-grow text-sm truncate"
              >
                Test 3
              </span>
            </div>
            <div
              class="hidden absolute items-center h-10 right-0 z-20"
              data-testid="tree-actions-3"
            />
          </div>
        </li>
      </ul>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
